File: typescript/text-analysis-tool/file-io.ts
import { readFile } from "fs/promises";

export async function readContentFromFile(filePath: string): Promise<string> {
  if (!filePath) {
    throw new Error("the filePath param is missing");
  }

  try {
    return await readFile(filePath, "utf-8");
  } catch (error) {
    throw error;
  }
}

File: golang/text-analysis-tool/file-io.go
package main

import (
	"errors"
	"os"
)

func ReadContentFromFile(filePath string) (string, error) {
	data, err := os.ReadFile(filePath)

	if err != nil {
		panic(err)
	}

	fileContent := string(data)

	if fileContent == "" {
		return "", errors.New("the file is empty")
	}

	return fileContent, nil
}
